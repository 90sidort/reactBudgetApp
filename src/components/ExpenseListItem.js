import React from 'react'
import { Link } from 'react-router-dom'


const ExpenseListItem = ({id, description, note, amount, createdAt}) => {
    return (
        <div>
            <Link to={`/edit/${id}`}>
                <h3>Description:{description}</h3>
            </Link> 
            <p>Amount:{amount}</p>
            <p>Note:{note}</p>
            <p>Creation Time:{createdAt}</p>
        </div>
    )
}

export default ExpenseListItem