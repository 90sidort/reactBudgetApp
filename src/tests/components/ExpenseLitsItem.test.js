import React from 'react'
import ExpenseListItem from '../../components/ExpenseListItem'
import { shallow } from 'enzyme'
import mockExpenses from '../fixtures/expenses'

test('Should render list item correctly', ()=> {
    const wrapper = shallow(<ExpenseListItem {...mockExpenses[0]} />)
    expect(wrapper).toMatchSnapshot()
})