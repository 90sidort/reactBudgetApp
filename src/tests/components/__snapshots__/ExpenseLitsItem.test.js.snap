// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render list item correctly 1`] = `
<div>
  <Link
    replace={false}
    to="/edit/1"
  >
    <h3>
      Description:
      First expense
    </h3>
  </Link>
  <p>
    Amount:
    100
  </p>
  <p>
    Note:
    TestNote1
  </p>
  <p>
    Creation Time:
    0
  </p>
</div>
`;
